### DISCUSSION TITLE
How can I run a batch file on uninstall with wix toolset 5?

### DISCUSSION THREAD

**Discussion by Lolle2000la** [wixtoolset/issues#8776] [Questions] [answered]
I need to run a batch file when uninstalling (ideally silently). I have tried the following, but it doesn't work.

```xml
<Binary Id="DeleteRegistryKeysBatch" SourceFile="delete_explorer_context_menu.bat" />

<CustomAction Id="RemoveImageSortKeys" BinaryRef="DeleteRegistryKeysBatch" ExeCommand="" Execute="deferred" Return="ignore" Impersonate="no" />

<InstallExecuteSequence>
  <Custom Action="EXECUTE_AFTER_FINALIZE" After="InstallFinalize" Condition="(AUTOSTART=1)" />
  <Custom Action="RemoveImageSortKeys" After="RemoveFiles" Condition="(NOT UPGRADINGPRODUCTCODE)"/>
</InstallExecuteSequence>
```

**Comment by thebruuu** [comment]
Did you try to run it in the Execommand  via a cmd or a start?
so your ExeCommand should be smthing like "cmd /c [#DeleteRegistryKeysBatch]"

**Reply by Lolle2000la** [reply]
I have tried your suggestion with no luck. If I leave the ExeCommand attribute empty and reference the script as a FileRef, then it does work though.

**Comment by Lolle2000la** [comment, marked as answer]
I have found a way, though it shows a command line window to the user, which isn't particularily desirable. First, I have added the script to the installation folder:

```xml
<Component Id="DeleteExplorerShortcuts" Guid="c601f684-d128-41b9-b3e6-153147942f67">
  <File Id="DeleteExplorerShortcutsScript" Name="delete_explorer_context_menu.bat"
        Source="delete_explorer_context_menu.bat" KeyPath="yes" />
</Component>
```

Then I have referenced it for the custom action instead:

```xml
<CustomAction Id="RemoveImageSortKeys" 
              FileRef="DeleteExplorerShortcutsScript" 
              ExeCommand="" 
              Execute="deferred" 
              Return="ignore" 
              Impersonate="no" />
```

The rest stays like above. This will now execute on uninstall (and if I understand correctly NOT on upgrades, which also triggers RemoveFiles).

**Reply by wmanning** [reply]
See Wix documentation: [Quiet execution custom actions](https://wixtoolset.org/docs/tools/wixext/quietexec/)
